src/db/data.rs

```diff
@@ -216,20 +216,27 @@ impl RawRecordData {
     }
 }
 
-impl EntryData for RawRecordData {
-    fn fields(&self) -> impl Iterator<Item = (&str, &str)> {
-        let (_, data_blocks) = self.split_blocks();
-        RawRecordFieldsIter {
-            remaining: data_blocks,
-        }
-    }
+macro_rules! entry_data_impl {
+    ($($raw:tt)*) => {
+        impl EntryData for $($raw)* {
+            fn fields(&self) -> impl Iterator<Item = (&str, &str)> {
+                let (_, data_blocks) = self.split_blocks();
+                RawRecordFieldsIter {
+                    remaining: data_blocks,
+                }
+            }
 
-    fn entry_type(&self) -> &str {
-        let (type_block, _) = self.split_blocks();
-        from_utf8(&type_block[1..]).unwrap()
-    }
+            fn entry_type(&self) -> &str {
+                let (type_block, _) = self.split_blocks();
+                from_utf8(&type_block[1..]).unwrap()
+            }
+        }
+    };
 }
```

Comment:
We can explicitly disallow an empty series of tokens.
```suggestion
macro_rules! entry_data_impl {
    ($($raw:tt)+) => {
        impl EntryData for $($raw)+ {
            fn fields(&self) -> impl Iterator<Item = (&str, &str)> {
                let (_, data_blocks) = self.split_blocks();
                RawRecordFieldsIter {
                    remaining: data_blocks,
                }
            }

            fn entry_type(&self) -> &str {
                let (type_block, _) = self.split_blocks();
                from_utf8(&type_block[1..]).unwrap()
            }
        }
    };
}
```